File: TicketBookingProject.Server/Models/Permission.cs
using System;
using System.Collections.Generic;

namespace TicketBookingProject.Server.Models;

public partial class Permission
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string Action { get; set; } = null!;

    public string Resource { get; set; } = null!;

    public string? Description { get; set; }

    public virtual ICollection<Role> Roles { get; set; } = new List<Role>();
}

File: ticketbookingproject.client/src/app/features/permission/models/permission.ts
export interface Permission {
  id: number;
  name: string;
  action: string;
  resource: string;
  description: string;
  roleStates: Record<number, boolean>; // { 1: true, 2: false, ... }
  isUpdating?: boolean;
}

export interface CreatePermissionDto {
  action: string;
  resource: string;
  description?: string;
}

export interface TogglePermissionDto {
  roleId: number;
  permissionId: number;
  isSelected: boolean;
}
export interface Role {
  id: number;
  name: string;
  description?: string;
  permissionCount: number;
}
